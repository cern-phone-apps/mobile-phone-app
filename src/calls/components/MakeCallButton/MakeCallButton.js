import React, { Component } from 'react';
import { Button, Icon } from 'react-native-elements';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { withNavigation } from 'react-navigation';

import { withPhoneService } from '../../providers/PhoneProvider/PhoneService';
import { logMessage } from '../../../common/utils/logging';
import ColorPalette from '../../../styles/ColorPalette';

const styles = StyleSheet.create({
  button: {
    marginTop: 10,
    marginBottom: 10,
    backgroundColor: ColorPalette.primary
  }
});

export class MakeCallButton extends Component {
  static propTypes = {
    phoneService: PropTypes.shape({ makeCall: PropTypes.func.isRequired })
      .isRequired,
    phoneNumber: PropTypes.string.isRequired,
    name: PropTypes.string
  };

  static defaultProps = {
    name: 'Unknown'
  };

  makeCall = () => {
    const { phoneService, phoneNumber, name } = this.props;

    logMessage(`Calling user ${phoneNumber}`);
    phoneService.makeCall(name, phoneNumber);
  };

  render() {
    return (
      <Button
        icon={<Icon name="phone" color="white" />}
        iconLeft
        title="Call this number"
        buttonStyle={[styles.button]}
        onPress={this.makeCall}
        testID="MakeCallButton"
      />
    );
  }
}

export default withNavigation(withPhoneService(MakeCallButton));
