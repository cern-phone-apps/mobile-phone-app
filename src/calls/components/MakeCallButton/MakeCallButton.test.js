import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { MakeCallButton } from './MakeCallButton';

const phoneNumber = '64446';
const name = 'nombre';
const phoneService = {
  makeCall: jest.fn((nameReceived, phoneReceived) => {
    expect(phoneNumber).toEqual(phoneReceived);
    expect(name).toEqual(nameReceived);
  })
};

/*
  We render the component
*/
test('Render the component', async () => {
  const { baseElement, queryByTestId } = render(
    <MakeCallButton
      phoneNumber={phoneNumber}
      phoneService={phoneService}
      name={name}
    />
  );

  expect(queryByTestId('MakeCallButton')).not.toBeNull();
  expect(baseElement).toMatchSnapshot();
});

/*
  We press the button
*/
test('Render the component and press the button', async () => {
  const { queryByTestId } = render(
    <MakeCallButton
      phoneNumber={phoneNumber}
      phoneService={phoneService}
      name={name}
    />
  );

  expect(queryByTestId('MakeCallButton')).not.toBeNull();
  const button = queryByTestId('MakeCallButton');
  fireEvent.press(button);
  expect(phoneService.makeCall).toHaveBeenCalledTimes(1);
});
