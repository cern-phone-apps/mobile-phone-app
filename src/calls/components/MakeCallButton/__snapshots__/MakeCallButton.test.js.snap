// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the component 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.3}
        disabled={false}
        iconLeft={true}
        testID="MakeCallButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#22529e",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
              "padding": 8,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                ÓÉç
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Call this number
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
